source/common/network/utility.cc

```diff
@@ -246,8 +260,10 @@ Address::InstanceConstSharedPtr Utility::getLocalAddress(const Address::IpVersio
         (ifa->ifa_addr->sa_family == AF_INET6 && version == Address::IpVersion::v6)) {
       const struct sockaddr_storage* addr =
           reinterpret_cast<const struct sockaddr_storage*>(ifa->ifa_addr);
-      ret = Address::addressFromSockAddr(
+      StatusOr<Address::InstanceConstSharedPtr> error_or_instance = Address::addressFromSockAddr(
           *addr, (version == Address::IpVersion::v4) ? sizeof(sockaddr_in) : sizeof(sockaddr_in6));
+      RELEASE_ASSERT(error_or_instance.ok(), error_or_instance.status().ToString());
```

Comment:
again I'm wondering if there was any try/catch on the stack here that previously would have caught an error.

Maybe for this case (and any others) where you are adding a new RELEASE_ASSERT, you could put ASSERT(!IsMainThread()). We know already it's invalid to have a try/catch on a worker thread so any thrown exception would crash anyway.

Also, can you use getAddressFromSockAddrOrDie here and other cases, rather than spelling out the RELEASE_ASSERT at each call-site?